MathPortal/MathPortal/Source/Custom/Custom Views/EquationView.swift

```diff
@@ -228,25 +228,40 @@ extension EquationView {
     
     static func generateFunction(_ inputViews: [EquationView], selectedColor: UIColor = Equation.defaultColor, color: UIColor = UIColor.black, scale: CGFloat = 1, type: Equation.ExpressionType, brackets: Bool = false) -> EquationView {
         guard inputViews.count > 0 else { return .Nil }
+        var text: String?
```

Comment:
This looks like you should use a switch. Also better to do

```
let text: String = {
    switch type...
}()
```

Funny thing is you can also use `let` as long as all path assign the value:

let text: String?
```
if type == .logarithm {
        text = "log"
} else {
        text = nil
}
```